### PR TITLE
Document `updates.bottlerocket.aws` endpoint

### LINKED ISSUE #336: Document utilization of `https://updates.bottlerocket.aws` endpoint for updates
The Bottlerocket updater API utilizes a public URL to query/check for updates: https://updates.bottlerocket.aws

This is burried pretty deep in the interlacing of our kubernetes operator, calling out to the bottlerocket `apiclient`, and then interfacing with the updater API. The usage of this URL may not be immediately clear to users (especially those with more restrictive egress firewalls)

### Issue or Feature Request:

We should document the usage of this public url and make note of the need for egress to this address.

### PR DESCRIPTION
**Pull request by jpmcb**
**Issue number:**

Closes https://github.com/bottlerocket-os/bottlerocket-update-operator/issues/336

**Description of changes:**

Documents the bottlerocket updater API's usage of `https://updates.bottlerocket.aws`

**Testing done:**

N/a

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Review by stmcginnis — APPROVED** [review]
Great to have - I know there are some users where this will be a common issue.

And no typos that I could find! ;)

File: README.md [hunk 1]
```diff
@@ -372,6 +372,29 @@ Then fetch the logs for that agent:
 kubectl logs brupop-agent-podname --namespace brupop-bottlerocket-aws 
 ```
 
+### Why are my bottlerocket nodes egressing to `https://updates.bottlerocket.aws`?
+
+The [Bottlerocket updater API](https://github.com/bottlerocket-os/bottlerocket/blob/develop/sources/updater/README.md)
+utilizes metadata from a public URL to query and check for updates.
+The URL is `https://updates.bottlerocket.aws` and the Bottlerocket's `apiclient` requires
```
Comment (webern) [inline README.md]:
```suggestion
The URL is `https://updates.bottlerocket.aws` and the Bottlerocket's update system requires
```

File: README.md [hunk 2]
```diff
@@ -372,6 +372,29 @@ Then fetch the logs for that agent:
...
+may not be able to reach this metadata endpoint
+and you may see failures when updates are available or an update status that appears "stuck".
+
+To troubleshoot and validate this, [access one of your Bottlerocket nodes on your cluster](https://github.com/bottlerocket-os/bottlerocket/blob/9adbe40c3fced559b2d05829e62902060c5ecc9c/README.md#exploration),
+and manually execute `apiclient update check`. If you see errors that indicate `Failed to check for updates`
+and network timeouts, ensure that your nodes on cluster have access to `https://updates.bottlerocket.aws`.
+
+If it is unacceptable to give your Bottlerocket nodes outside network access,
+you may consider creating a self managed proxy within the cluster
+that can periodically scrape the update metadata information from `https://updates.bottlerocket.aws`.
```
Comment (webern) [inline README.md]: metadata?
They would need to clone the metadata and the targets to make use of the update repo.

```suggestion
that can periodically scrape the update ? information from `https://updates.bottlerocket.aws`.
```

**Review by webern — COMMENTED** [review]
The documentation you've written assumes that the user is not building their own variant and maintaining their own TUF repo, but maybe that's fine.

Nits:
- apiclient doesn't contact the update repo, technically it's `updog`
- I'm not sure what you mean by scraping metadata. I think they would need to duplicate the TUF repo in question.

**Comment by jpmcb** [comment]
> The documentation you've written assumes that the user is not building their own variant and maintaining their own TUF repo, but maybe that's fine.

True. This doc assumes the user isn't building / serving / or using their own TUF repos. That seems like a pretty advanced use case, but maybe worth mentioning here too.

> I'm not sure what you mean by scraping metadata. I think they would need to duplicate the TUF repo in question.

Essentially yes, the user would need to have a proxy that scrapes the TUF repo and serves that locally on cluster. I've seen this sort of thing done before via an Nginx reverse proxy or squid proxy.

I'll update this to be more clear - thanks for the feedback Matt!

**Comment by jpmcb** [comment]
Force pushed to address above comments!

**Review by webern — APPROVED** [review]

**Pull request merged**